src/main/kotlin/org/rucca/cheese/task/TaskMembershipService.kt

```
@@ -262,6 +570,32 @@ class TaskMembershipService(
         if (taskMembershipRepository.existsByTaskIdAndMemberId(task.id!!, memberId))
             return AlreadyBeTaskParticipantError(task.id!!, memberId)
 
+        // Check if user has real name info if required
+        if (task.requireRealName == true) {
+            when (task.submitterType) {
+                TaskSubmitterType.USER -> {
+                    // For individual participants, check if they have real name info
+                    try {
+                        userRealNameService.getUserIdentity(memberId)
+                    } catch (e: NotFoundError) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)